docs/graphx-programming-guide.md

```diff
@@ -708,7 +708,9 @@ messages remaining.
 > messaging function.  These constraints allow additional optimization within GraphX.
 
 The following is the type signature of the [Pregel operator][GraphOps.pregel] as well as a *sketch*
-of its implementation (note calls to graph.cache have been removed):
+of its implementation (note: to avoid stackOverflowError due to long lineage chains, graph and 
```

Comment:
Personally, I don't think we need to include the checkpointer in the implementation sketch. I think it's more an implementation detail than an essential part of the algorithm. I think it's enough to simply document that GraphX's implementation of pregel includes checkpointing to avoid unbounded RDD lineages.